src/visualization/plot_recipes.jl

```diff
@@ -478,21 +478,29 @@ function to `solution_variables`.
 twice the number of solution DG nodes are used for visualization, and if set to `0`,
 exactly the number of nodes in the DG elements are used.
 
-
+When visualizing data from a two-dimensional simulation, a 1D stripe is extracted for plotting.
+`stripe_axis` specifies the axis stripped from the plane and may be `:x`, or `:y`. The
+point on the stripe axis where it intersects with the orthogonal axis is given in `stripe_axis_intercept`.
+Both of these values are ignored when visualizing 1D data.
 
 !!! warning "Experimental implementation"
     This is an experimental feature and may change in future releases.
 """
-function PlotData1D(u, semi; solution_variables=nothing, nvisnodes=nothing)
+function PlotData1D(u, semi; solution_variables=nothing, nvisnodes=nothing, stripe_axis=:x, stripe_axis_intercept=0)
 
   mesh, equations, solver, cache = mesh_equations_solver_cache(semi)
-  @assert ndims(mesh) in (1) "unsupported number of dimensions $ndims (must be 1)"
+  @assert ndims(mesh) in (1, 2) "unsupported number of dimensions $ndims (must be 1 or 2)"
   solution_variables_ = digest_solution_variables(equations, solution_variables)
 
   variable_names = SVector(varnames(solution_variables_, equations))
   original_nodes = cache.elements.node_coordinates
 
   unstructured_data = get_unstructured_data(u, semi, solution_variables_)
+
+  if ndims(mesh) == 2
+    original_nodes, unstructured_data = unstructured_1d_to_2d(original_nodes, unstructured_data, stripe_axis, stripe_axis_intercept)
+  end
```

Comment:
I don't like overwriting variables with a different meaning, especially since it is *not* the original nodes anymore. Maybe you can move the `if ndims(mesh) == 2` to the very end, and either first call `unstructured_1d_to_2d` and pass the result to `PlotData1D`, or just return the current `PlotData1D` construction

Reply:
I changed this so there is no overwriting anymore.